### ISSUE TITLE
error while loading shared libraries: libcriterion.so.3: cannot enable executable stack as shared object requires: Invalid argument

### ISSUE THREAD

**Issue by hlimbo** [Microsoft/BashOnWindows#2562]
Hi,

I am using the Criterion, which is a C Unit Testing Framework (https://github.com/Snaipe/Criterion). After I finished installing the Criterion package, I wanted to test out if the package was installed properly by typing up the following:

```C
 #include <criterion/criterion.h>
Test(simple,the_test)
{ 
   cr_assert(0,"Hello World");
}
```
The file compiles regularly as expected using:
```
gcc -o test simple_test.c -lcriterion
```
However whenever the I attempt to run the executable `test` it gives the following error:
```
./test: error while loading shared libraries: libcriterion.so.3: cannot enable executable stack as shared object requires: Invalid argument
```

I have already tried a possible solution found here: #https://github.com/Microsoft/BashOnWindows/issues/916 but has not worked for me.

**Comment by therealkenc** [comment]
> `ImportError: /usr/local/lib/python2.7/dist-packages/cryptography/hazmat/bindings/_openssl.so: cannot enable executable stack as shared object requires: Invalid argument`

I'm guessing everyone's crypto libraries must have levelled recently (for some value of recent). #2561, #2553, #916, #442, #286 et al.

**Comment by rowanG077** [comment]
Did you try clearing the executeable stack flag? `execstack -c <PathToLibrary>/libcriterion.so.3`

**Comment by hlimbo** [comment]
Yes I tried do that but after I recompile and try to rerun the program, the program hangs without any error or warning messages.

**Comment by rowanG077** [comment]
Then it most likely actually requires executable stack. Which won't be supported by design on WSL. To be sure that is what is happening you could try to build your project on an actual ubuntu machine where you cleared the executable flag on the library and see if it works there. See: https://github.com/Microsoft/BashOnWindows/issues/286

**Comment by therealkenc** [comment]
> Which won't be supported by design on WSL.

Yeah the entire ecosystem is busy implementing `if (workaround_wsl)` as we speak. Like [here](https://github.com/dhermes/bezier/commit/8fe1fdbfc38f739d02bc62ea11af948b3d58fe9e), posted to [biezer](https://bezier.readthedocs.io/en/latest/index.html) just four days ago.

**Comment by rowanG077** [comment]
Well if wsl workarounds arent good and the teams implementing it are wrong. They should just not use libraries that are still using executable stack. It`s a huge security risk.

**Comment by hlimbo** [comment]
I've rebuilt my project on an actual Ubuntu machine and the library I'm using works as intended. Its unfortunate I can't use libraries that require the use of an executable stack.

**Comment by rowanG077** [comment]
You cleared the executable flag on your actual ubuntu machine and it everything works? If that is true then this is indeed a bug.

**Comment by hlimbo** [comment]
I did not need to clear the executable flag on the ubuntu machine. Everything works as intended when I install the library regularly.
